File: Batalla_Poo_CPP/src/Participante.cpp
#include "Participante.h"

Participante::Participante(){}

void Participante::setNombre(string nombre)
{
    this->nombre = nombre;
}

string Participante::getNombre()
{
    return this->nombre;
}

void Participante::setVida(int vida)
{
    this->vida = vida;
}

int Participante::getVida()
{
    return this->vida;
}

//void Participante::disparar(Participante enemigo){}

int Participante::quitarVida(int valor)
{
     setVida(getVida() - valor);
     return getVida();
}

bool Participante::estaVivo()
{
    return (getVida()>0);
}

int Participante::danioParaTanque() const {}
int Participante::danioParaAvion(){}
int Participante::danioParaSoldado(){}

File: Batalla_Poo_Ruby/participante.rb
class Participante

    attr_accessor :nombre, :vida

    def initialize()
    end

    def getTipo
    end

    def danio(valor)
        self.vida = self.vida - valor
        return self.vida
    end

    def estaVivo?
        self.vida > 0
    end

    def disparar(enemigo)
    end

    def danioParaTanque()
    end

    def danioParaSoldado()
    end

    def danioParaAvion()
    end

end